lib/addons/karpenter/index.ts

```diff
@@ -54,23 +69,66 @@ export class KarpenterAddOn extends HelmAddOn {
         const cluster = clusterInfo.cluster;
         const values = { ...this.props.values ?? {} };
 
+        const provisionerSpecs = this.options.ProvisionerSpecs || {};
+        const subnetTags = this.options.SubnetTags || {};
+        const sgTags = this.options.SecurityGroupTags || {};
+
         // Tag VPC Subnets
-        tagSubnets(cluster.stack, cluster.vpc.privateSubnets, `kubernetes.io/cluster/${clusterInfo.cluster.clusterName}`,"1");
+        if (subnetTags){
+            Object.entries(subnetTags).forEach(
+                ([key,value]) => {
+                    tagSubnets(cluster.stack, cluster.vpc.privateSubnets, key, value);
+            });
+        }
+        
+        // Tag VPC Security Group
+        if (sgTags){
+            Object.entries(sgTags).forEach(
+                ([key,value]) => {
+                    tagSecurityGroup(cluster.stack, cluster.clusterSecurityGroupId, key,value);
+            });
+        }
+        
+        const conditions = new CfnJson(cluster, "ConditionPlainJson", {
```

Comment:
Statements from line 92 to line 116 is just a registration of an IRSA account the hard way. 

Why not just use standard IRSA service account support in CDK, i.e. 

```
const sa = cluster.addServiceAccount(roleName, opts);
sa.role.addManagedPolicy()...
```

This automatically creates OIDC provider and trust relationship.

Reply:
The problem is the SA role needs to follow a naming convention in the form of `${clusterName}-karpenter`. addServiceAccount autogenerates the role name. This is something I've considered opening an issue for against CDK.

Reply:
Addressed - used our own `createServiceAccount`